pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -2,20 +2,28 @@ package ipam
 
 import (
 	"fmt"
+	"net"
 	"strings"
 
 	compute "google.golang.org/api/compute/v1"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/labels"
 	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
 	"k8s.io/klog/v2"
+	netutils "k8s.io/utils/net"
 )
 
-// PerformMultiNetworkCIDRAllocation allots pod CIDRs for all the networks that a node is connected to. It handles IPv6 only for default-network for now.
-func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, interfaces []*compute.NetworkInterface) (defaultNwCIDRs []string, northInterfaces networkv1.NorthInterfacesAnnotation, additionalNodeNetworks networkv1.MultiNetworkAnnotation, err error) {
+// PerformMultiNetworkCIDRAllocation allots pod CIDRs for all the networks that
+// a node is connected to. It handles IPv6 only for default-network for now.
+// Any changes to API should be reflected in the 'node' variable.
```

Comment:
Maybe something more explicit
```
PerformMultiNetworkCIDRAllocation receives the existin Node objects and its interfaces, and is
updated with the corresponding annotations for MultiNetwork and NorthInterfaces and the
capacity for the additional networks
```
the test needs rewording but the idea is that the next person will not need to read the code of the function